haystack/components/converters/html.py

```diff
@@ -96,28 +99,56 @@ def run(
         documents = []
         meta_list = normalize_metadata(meta=meta, sources_count=len(sources))
 
-        extractor_class = getattr(extractors, self.extractor_type)
-        extractor = extractor_class(raise_on_failure=False)
+        # Use all extractor types, ensuring user chosen extractor is first, preserve order, avoid duplicates
+        extractors_list = list(
+            dict.fromkeys(
+                [
+                    self.extractor_type,  # User chosen extractor is always tried first
+                    "DefaultExtractor",
+                    "ArticleExtractor",
+                    "ArticleSentencesExtractor",
+                    "LargestContentExtractor",
+                    "CanolaExtractor",
+                    "KeepEverythingExtractor",
+                    "NumWordsRulesExtractor",
+                ]
+            )
+        )
 
         for source, metadata in zip(sources, meta_list):
             try:
                 bytestream = get_bytestream_from_source(source=source)
-            except Exception as e:
-                logger.warning("Could not read {source}. Skipping it. Error: {error}", source=source, error=e)
+            except Exception:
+                logger.warning(f"Could not read {source}. Skipping it.")
                 continue
-            try:
-                file_content = bytestream.data.decode("utf-8")
-                text = extractor.get_content(file_content)
-            except Exception as conversion_e:
+
+            text = None
+            for extractor_idx, extractor_name in enumerate(extractors_list):
+                if extractor_idx > 0 and not self.try_others:
+                    break
+                extractor_class = getattr(extractors, extractor_name)
+                extractor = extractor_class(raise_on_failure=False)
```

Comment:
as mentioned above, this part can be simplified if the `extractors_list` only contains the actual extractors to try:
we could probably get rid of `enumerate` and `extractor_idx`.